dkms.in

```diff
@@ -2206,14 +2207,17 @@ autoinstall() {
     # a list of modules and their latest version.
     while read status mvka; do
         IFS='/' read m v k a <<< "$mvka"
-        if [[ ("$(VER "$v")" > "$(VER "${latest["$m"]}")") ]]; then
+        if [[ -z "${latest["$m"]}" ]]; then
+            known_modules[${#known_modules[@]}]="$m"
+            latest["$m"]="$v"
+        elif [[ ("$(VER "$v")" > "$(VER "${latest["$m"]}")") ]]; then
             latest["$m"]="$v"
         fi
     done < <(module_status)
 
     # Walk through our list of known modules, and create
     # a list of modules that need to be reinstalled.
-    for m in "${!latest[@]}"; do
+    for m in "${known_modules[@]}"; do
         v="${latest["$m"]}"
```

Comment:
It's no immediately obvious why the order matters at this point. All we do is gather a list of to install modules and their respective deps.

Can you add some context in the commit message?

Reply:
mainly for deterministic tests. if autoinstall is going to build more than one module (successful or not), order does matter for the output